File: Python3/프로그래머스/1/135808. 과일 장수/과일 장수.py
def solution(k, m, score):
    # 1. 점수를 내림차순으로 정렬
    score.sort(reverse=True)
    
    # 2. 최대 이익을 계산
    max_profit = 0
    for i in range(0, len(score) - m + 1, m):
        # 그룹의 최저 점수는 i + m - 1번째 사과의 점수
        min_score_in_box = score[i + m - 1]
        max_profit += min_score_in_box * m
    
    return max_profit

File: C++/프로그래머스/1/135808. 과일 장수/과일 장수.cpp
#include <vector>
#include <algorithm>

using namespace std;

int solution(int k, int m, vector<int> score) {
    // 사과 점수를 내림차순으로 정렬
    sort(score.rbegin(), score.rend());
    
    int totalProfit = 0;
    int count = 0;

    // 상자 단위로 처리
    for (int s : score) {
        count++;
        if (count == m) {
            totalProfit += s * m;  // 상자에 담긴 사과의 점수 * m (상자에 담긴 사과 개수)
            count = 0;  // 상자 포장이 끝났으므로 카운트 초기화
        }
    }
    
    return totalProfit;
}
